src/NuGet.Indexing/Extraction/CatalogPackageMetadataExtraction.cs

```diff
@@ -55,18 +57,52 @@ private class Extractor
                 AddString(MetadataConstants.ReleaseNotesPropertyName);
                 AddString(MetadataConstants.CopyrightPropertyName);
                 AddString(MetadataConstants.LanguagePropertyName);
-                AddString(MetadataConstants.LicenseUrlPropertyName);
                 AddString(MetadataConstants.PackageHashPropertyName);
                 AddString(MetadataConstants.PackageHashAlgorithmPropertyName);
                 AddString(MetadataConstants.PackageSizePropertyName);
                 AddString(MetadataConstants.CatalogMetadata.RequiresLicenseAcceptancePropertyName, MetadataConstants.RequiresLicenseAcceptancePropertyName);
 
+                AddLicenseUrl();
                 AddFlattenedDependencies();
                 AddSupportedFrameworks();
 
                 return _metadata;
             }
 
+            private void AddLicenseUrl()
+            {
+                if (_galleryBaseAddress != null && _catalog[MetadataConstants.LicenseExpressionPropertyName] != null)
+                {
+                    _metadata[MetadataConstants.LicenseUrlPropertyName] = GetOverWriteLicenseUrl();
+                } else if (_galleryBaseAddress != null && _catalog[MetadataConstants.LicenseFilePropertyName] != null)
+                {
+                    _metadata[MetadataConstants.LicenseUrlPropertyName] = GetOverWriteLicenseUrl();
+                } else
+                {
+                    AddString(MetadataConstants.LicenseUrlPropertyName);
+                }
+            }
+
+            private string GetOverWriteLicenseUrl()
```

Comment:
We should have a similar method in Ctalog2Registration. Can we share the code?

Reply:
We have a  sharing license helper to get license Url now.